### PR TITLE
Add http_forwarder skeleton

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Add skeleton for `http_forwarder` extension that forwards HTTP requests to a specified target.

**Testing:** Tests added.

**Documentation:** README added.

### CODE REVIEW

**Comment by asuresh4** [comment]
PTAL @jrcamp @bogdandrutu

File: extension/httpforwarder/config.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// Config defines configuration for http forwarder extension.
+type Config struct {
+	configmodels.ExtensionSettings `mapstructure:",squash"`
+	confighttp.HTTPServerSettings  `mapstructure:",squash"`
```
Comment (bogdandrutu) [inline extension/httpforwarder/config.go:25]: I would also name this "Incoming" and the client "Outgoing", what do you think? same in config probably to make it clear for user.

File: extension/httpforwarder/extension.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	// TODO: Start HTTP server
+	return nil
+}
+
+func (h httpForwarder) Shutdown(_ context.Context) error {
+	// TODO: Shutdown HTTP server
+	return nil
+}
+
+func newHTTPForwarder(config *Config, logger *zap.Logger) (component.ServiceExtension, error) {
```
Comment (bogdandrutu) [inline extension/httpforwarder/extension.go:49]: This should be removed from skeleton, only a complete no-op should be returned.

File: extension/httpforwarder/factory.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		},
+	}
+}
+
+func createExtension(
+	_ context.Context,
+	params component.ExtensionCreateParams,
+	cfg configmodels.Extension,
+) (component.ServiceExtension, error) {
+	return newHTTPForwarder(cfg.(*Config), params.Logger)
```
Comment (bogdandrutu) [inline extension/httpforwarder/factory.go:63]: Maybe just return nil for the moment.

File: extension/httpforwarder/config.go (hunk 1 above)
Comment (asuresh4) [reply extension/httpforwarder/config.go]: Updated.

File: extension/httpforwarder/extension.go (hunk 2 above)
Comment (asuresh4) [reply extension/httpforwarder/extension.go]: Removed.

File: extension/httpforwarder/factory.go (hunk 3 above)
Comment (asuresh4) [reply extension/httpforwarder/factory.go]: Done.

File: extension/httpforwarder/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  http_forwarder:
+    incoming:
+      endpoint: localhost:7070
+    outgoing:
+      endpoint: http://target/
+      headers:
+        otel_http_forwarder: dev
+      timeout: 5s
+```
+
```
Comment (bogdandrutu) [inline extension/httpforwarder/README.md:27]: Some extra whitespace I think here

File: extension/httpforwarder/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* `outgoing`: HTTP config settings to use for forwarding requests.
+  * `endpoint` (**required**): The target to which requests should be forwarded to.
+  * `headers` (default: `nil`): Additional headers to be added to all requests passing through the extension.
+  * `timeout` (default: `10s`): How long to wait for each request to complete.
+
+### Example
+
+```yaml
+  http_forwarder:
+    incoming:
```
Comment (bogdandrutu) [inline extension/httpforwarder/README.md:19]: Not blocking, do you think actually ingress and egress make more sense?

**Review by bogdandrutu — APPROVED** [review]

File: extension/httpforwarder/README.md (hunk 4 above)
Comment (asuresh4) [reply extension/httpforwarder/README.md]: Removed.

File: extension/httpforwarder/README.md (hunk 5 above)
Comment (asuresh4) [reply extension/httpforwarder/README.md]: yes, it probably makes more sense. Renamed them.

**Pull request merged**